src/components/ListItem.jsx

```diff
@@ -4,17 +4,79 @@ import React, { useState } from 'react';
 // LOCAL IMPORTS
 import './ListItem.css';
 import { updateItem } from '../api/firebase.js';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
 
-export function ListItem({ name, listToken, dateLastPurchased, itemId }) {
+export function ListItem({
+	name,
+	listToken,
+	dateLastPurchased,
+	itemId,
+	totalPurchases,
+	dateCreated,
+	dateNextPurchased,
+}) {
+	const now = Date.now() / 1000; //current time in seconds
 	// SET STATES
 	const [isChecked, setIsChecked] = useState(
-		Date.now() / 1000 - dateLastPurchased?.seconds < 60 * 60 * 24,
+		now - dateLastPurchased?.seconds < 60 * 60 * 24,
 	);
 
+	// HELPER FUNCTIONS
+	const getDaysSinceLastTransaction = (dateLastPurchased, dateCreated) => {
```

Comment:
In general I would move all of the helper functions that you can into `utils`, and where possible date conversion values should be set as constants in that file ex: `60 * 60 * 24`